docs/graalvm-compatibility-in-ballerina-libraries.md

```diff
@@ -0,0 +1,185 @@
+
+Ballerina runtime, [standard libraries](https://ballerina.io/learn/ballerina-specifications/#standard-library-specifications), and the Ballerina extended modules are GraalVM-compatible. Therefore packages developed only using these libraries are also GraalVM-compatible. Furthermore, Ballerina reports warnings when the GraalVM build is executed for a project with GraalVM-incompatible packages.
+
+```
+**********************************************************************************
+* WARNING: Package is not verified with GraalVM.                                 *
+**********************************************************************************
+
+The GraalVM compatibility property has not been defined for the package
+'<package-name>. This could potentially lead to compatibility issues with GraalVM.
+
+To resolve this warning, please ensure that all Java dependencies of this
+package are compatible with GraalVM. Subsequently, update the Ballerina.toml
+file under the section '[platform.<java*>]' with the attribute
+'graalvmCompatible = true'.
+
+**********************************************************************************
+```
+
+### Analyzing the code base for Java dynamic features
+Since the native-image tool assumes the closed-world assumption, the following [dynamic features](https://www.graalvm.org/jdk17/reference-manual/native-image/dynamic-features/) of Java should be handled explicitly through configuration files.
+- [Accessing Resources](https://www.graalvm.org/jdk17/reference-manual/native-image/dynamic-features/Resources/)
+- [Certificate Management](https://www.graalvm.org/jdk17/reference-manual/native-image/dynamic-features/CertificateManagement/)
+- [Dynamic Proxy](https://www.graalvm.org/jdk17/reference-manual/native-image/dynamic-features/DynamicProxy/)
+- [Java Native Interface (JNI)](https://www.graalvm.org/jdk17/reference-manual/native-image/dynamic-features/JNI/)
+- [JCA Security Services](https://www.graalvm.org/jdk17/reference-manual/native-image/dynamic-features/JCASecurityServices/)
+- [Reflection](https://www.graalvm.org/jdk17/reference-manual/native-image/dynamic-features/Reflection/)
+- [URL Protocols](https://www.graalvm.org/jdk17/reference-manual/native-image/dynamic-features/URLProtocols/)
+
+The analysis should be also done on the third-party libraries used in the module. Some third-party libraries might be already GraalVM compatible, this should be verified with the third-party library owners. You can also refer to the configurations for some of the commonly used libraries in [graalvm-reachability-metadata repository](https://github.com/oracle/graalvm-reachability-metadata/tree/master/metadata).
+
+### Test a sample application with GraalVM
+#### Prerequisites
+1. Latest [Ballerina Swan Lake](https://ballerina.io/downloads/) distribution
+   >**Note:** If you are using macOS with an ARM64 processor, then, install Ballerina using the [ARM64 installer](https://ballerina.io/downloads/).
```

Comment:
Do we need this note? Each OS and Architecture should follow this, right?

Reply:
This is added since both x64 and arm64 macOS installers works with ballerina. The issue raises only when we do GraalVM native build. Thats why added this as a specific note.